### PR TITLE
Improve security by inverting policy. Adding minor features.

### PR DESCRIPTION
**Pull request by zokradonh**
- If we deny all requests and only allow the API endpoints that are allowed then we are more secure if there are any new APIs coming out in the future => inverted allow/deny policy
- Added the experimental session API endpoint.
- Often some containers only need to restart/reload another container. Therefore I added a permission explicitly for that.

### CODE REVIEW

File: haproxy.cfg [hunk 1]
```diff
@@ -42,26 +42,29 @@ backend dockerbackend
...
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/images } { env(IMAGES) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/info } { env(INFO) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/networks } { env(NETWORKS) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/nodes } { env(NODES) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/_ping } { env(PING) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/plugins } { env(PLUGINS) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/post } { env(POST) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/secrets } { env(SECRETS) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/services } { env(SERVICES) -m bool }
+    http-request allow if { path,url_dec -m reg -i ^(/v[\d\.]+)?/session } { env(SESSION) -m bool }
```
Comment (Yajo) [inline haproxy.cfg]: This endpoint doesn't exist as per docs.

File: Dockerfile [hunk 2]
```diff
@@ -18,6 +19,7 @@ ENV AUTH=0 \
     POST=0 \
     SECRETS=0 \
     SERVICES=0 \
+    SESSION=0 \
```
Comment (Yajo) [inline Dockerfile]: :fire:

File: README.md [hunk 3]
```diff
@@ -135,6 +135,7 @@ does not need.
 - `NODES`
 - `PLUGINS`
 - `SERVICES`
+- `SESSION`
```
Comment (Yajo) [inline README.md]: :fire:

File: haproxy.cfg (hunk 1 above)
Comment (zokradonh) [reply haproxy.cfg]: Don't know if it is active by default but I have it from the docs:
https://docs.docker.com/engine/api/v1.37/#tag/Session-(experimental)

Comment (Yajo) [reply haproxy.cfg]: Oh I didn't see it. Thanks!

**Pull request merged**